statement_test.go

```diff
@@ -497,3 +498,43 @@ func TestGetQueryID(t *testing.T) {
 		t.Fatalf("failed to prepare statement. err: %v", err)
 	}
 }
+
+func TestAsyncMode(t *testing.T) {
```

Comment:
What does code coverage look like? Feels like there aren't enough test cases.

Reply:
It's at 80.7% with the push I just made.

Reply:
Got it, thanks. The main thing I'm worried about is making sure the edge cases are covered in tests and coverage helps highlight those.